### PR TITLE
docs: More Java extension documentation

### PR DESCRIPTION
**Pull request by notpeter**
Follow up of: https://github.com/zed-industries/zed/pull/19113

cc: @valentinegb @ABckh @HeavySnowJakarta

Release Notes:

- N/A

### CODE REVIEW

File: docs/src/languages/java.md [hunk 1]
```diff
@@ -40,6 +44,8 @@ We recommend you install one or the other and not both.
 
 ## Settings / Initialization Options
 
+For available `initialization_options` please see the [Initialize Request section of the Eclipse.jdt.ls Wiki](https://github.com/eclipse-jdtls/eclipse.jdt.ls/wiki/Running-the-JAVA-LS-server-from-the-command-line#initialize-request).
+
 See [JDTLS Language Server Settings & Capabilities](https://github.com/eclipse-jdtls/eclipse.jdt.ls/wiki/Language-Server-Settings-&-Capabilities) for a complete list of options.
```
Comment (valentinegb) [inline docs/src/languages/java.md:49]: Like I said earlier, this page is mostly related to compilation options and not initialization options, so this line should be removed.

Comment (notpeter) [reply docs/src/languages/java.md]: Thanks. I'm two decades out from the programming full time and so this is all very theoretical for me. :P

File: docs/src/languages/java.md
Comment (valentinegb) [inline docs/src/languages/java.md:1]: Wrong line!

Comment (notpeter) [reply docs/src/languages/java.md:1]: LOL. Put up a PR and I'll merge it.

Comment (valentinegb) [reply docs/src/languages/java.md:1]: Will do, you'll have to give me a few minutes though since I just got home, sorry.

**Pull request merged**